packages/store/addon/-private/identifiers/cache.ts

```diff
@@ -343,6 +343,15 @@ export class IdentifierCache {
     let newId = coerceId(data.id);
     let existingIdentifier = detectMerge(this._cache.types, identifier, data, newId, this._cache.lids);
 
+    if (!existingIdentifier) {
+      if (identifier.type !== data.type && data.type) {
```

Comment:
There's `if (existingIdentifier)` statement below, maybe it would be reasonable to have this as `else if ` after that?

I'd also prefer to add a comment about what sort of case this is meant to deal with, but totally up to you folks

Reply:
1. We still want the other statement to run, in fact the whole point of doing this here is to get line 357 
`      identifier = this._mergeRecordIdentifiers(keyOptions, identifier, existingIdentifier, data, newId as string);` to perform the merge

Reply:
2. Great point re: comment, adding that and merging

Reply:
OMG makes total sense. Thank you